client/src/apis/chessClient.sockets.js

```diff
@@ -0,0 +1,31 @@
+import IO from 'socket.io-client';
+
+let socket;
+
+const createConnection = () => {
+  socket = IO.connect('http://localhost:3001');
+};
+
+const receiveGameData = (cb) => {
+  socket.on('gameData', (gameData) => {
+    cb(gameData);
+  });
+};
+
+
+const sendMove = (moveObj) => {
+  socket.emit('pieceMoved', moveObj);
+};
+
+const receiveMove = (cb) => {
+  socket.on('movePiece', (moveObj) => {
+    cb(moveObj);
+  });
+};
+
+export {
+  receiveGameData,
+  createConnection,
```

Comment:
if you are exposing createConnection endpoint, you may as well return the connection itself as function response.

Reply:
You can ideally expose an object as a response to create connection, which includes other methods, and you can directly call methods on it. Not sure how much is it applicable here. Also, make sure there is always a counter disconnect connection as part of create connection. And disconnect connection on server termination. Some connections inherently support this though.

Reply:
Not sure what you mean by createConnection endpoint. It is just a function that when called establishes socket connection between client and server.

I have added a function to emit disconnect event to the server whenever the game ends or client disconnects.